packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -460,11 +477,51 @@ abstract class AssetBundleImageProvider extends ImageProvider<AssetBundleImageKe
   ///
   /// This function is used by [load].
   @protected
-  Future<ui.Codec> _loadAsync(AssetBundleImageKey key) async {
+  Future<ui.Codec> _loadAsync(AssetBundleImageKey key, DecoderCallback decode) async {
     final ByteData data = await key.bundle.load(key.name);
     if (data == null)
       throw 'Unable to read data';
-    return await PaintingBinding.instance.instantiateImageCodec(data.buffer.asUint8List());
+    return await decode(data.buffer.asUint8List());
+  }
+}
+
+class _SizeAwareCacheKey {
+  const _SizeAwareCacheKey(this._providerCacheKey, this._width, this._height);
+
+  final dynamic _providerCacheKey;
+  final int _width;
+  final int _height;
+}
+
+/// Re-sizes the cache of the image provided to the specified size.
+class ResizedImage extends ImageProvider<_SizeAwareCacheKey> {
+  /// Creates an object that re-sizes the cache of the image to the specified size.
+  ///
+  /// The image cached and returned through the [ImageStreamCompleter] will be
```

Comment:
that's not actually true. what it does is decode the image at that dimension, it doesn't resize it.

Reply:
How are decoding at a specific dimension and resizing to a specific dimension different?

Reply:
We can display at a different resolution than stored in the cache. For example, if someone wanted to display an image at 10x10, 20x20 and 100x100 with native 4000x4000, you can resize to 100x100 and reuse same cached image for each display.

Reply:
Is this correct?

1. cached size -> this is the size at which we store the image in the cache, so the size at which we decode the image.
2. resized size -> the size at which we display the image.

If  yes, we should try to convey this to the user somehow. If I was using this API, i would think resizing the image to a lower dimension will make the image memory footprint smaller too.